core/server/src/test/java/alluxio/master/file/meta/InodeFileTest.java

```diff
@@ -167,4 +174,214 @@ public void permissionStatus() {
     Assert.assertEquals(new Permission(TEST_PERMISSION).applyFileUMask().getMode().toShort(),
         inode1.getMode());
   }
+
+  /**
+   * Tests the {@link Inode#lockRead()} and {@link Inode#unlockRead()} methods.
+   */
+  @Test
+  public void lockRead() {
+    InodeFile inode1 = createInodeFile(1);
+    Assert.assertFalse(inode1.isReadLocked());
+    Assert.assertFalse(inode1.isWriteLocked());
+    inode1.lockRead();
+    Assert.assertTrue(inode1.isReadLocked());
+    Assert.assertFalse(inode1.isWriteLocked());
+    inode1.unlockRead();
+    Assert.assertFalse(inode1.isReadLocked());
+    Assert.assertFalse(inode1.isWriteLocked());
+  }
+
+  /**
+   * Tests the {@link Inode#lockReadAndCheckParent(Inode)} method.
+   */
+  @Test
+  public void lockReadAndCheckParent() throws Exception {
+    InodeFile inode1 = createInodeFile(1);
+    InodeDirectory dir1 = createInodeDirectory();
+    inode1.setParentId(dir1.getId());
+    inode1.lockReadAndCheckParent(dir1);
+    Assert.assertTrue(inode1.isReadLocked());
+    inode1.unlockRead();
+  }
+
+  /**
+   * Tests the {@link Inode#lockReadAndCheckParent(Inode)} method fails when the parent is
+   * not consistent.
+   */
+  @Test
+  public void lockReadAndCheckParentInvalid() throws Exception {
+    mExpectedException.expect(InvalidPathException.class);
+    mExpectedException.expectMessage(ExceptionMessage.PATH_INVALID_CONCURRENT_RENAME.getMessage());
+    InodeFile inode1 = createInodeFile(1);
+    InodeDirectory dir1 = createInodeDirectory();
+    inode1.setParentId(dir1.getId() - 1);
+    inode1.lockReadAndCheckParent(dir1);
+  }
+
+  /**
+   * Tests the {@link Inode#lockReadAndCheckFullPath(Inode, String)} method.
+   */
+  @Test
+  public void lockReadAndCheckFullPath() throws Exception {
```

Comment:
if we merge `inode.lockReadAndCheckFullPath()` to `inodeList.lockReadAndCheckFullPath()`, we will then test this method there, which seems makes more sense as we could assert the lock state on the list